web/src/shared/components/ThemeProvider.jsx

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react'
+import { useMediaQuery } from 'react-responsive'
+
+const ThemeContext = React.createContext()
+
+const ThemeProvider = (props) => {
+    const preferredMode = useMediaQuery({ query: '(prefers-color-scheme: dark)' })
+        ? 'dark-mode'
+        : 'light-mode'
+    const storedTheme = localStorage.getItem('theme') || 'system-mode'
+    const savedTheme = storedTheme !== 'system-mode' ? storedTheme : preferredMode
+    const [theme, setTheme] = React.useState(savedTheme)
+
+    const toggleTheme = (type) => {
+        if (type === 'light-mode') {
+            localStorage.setItem('theme', 'light-mode')
+            document.documentElement.setAttribute('data-theme', 'light-mode')
+            setTheme('light-mode')
+        } else if (type === 'dark-mode') {
+            localStorage.setItem('theme', 'dark-mode')
+            document.documentElement.setAttribute('data-theme', 'dark-mode')
+            setTheme('dark-mode')
+        } else {
+            localStorage.setItem('theme', 'system-mode')
+            document.documentElement.setAttribute('data-theme', preferredMode)
+            setTheme(preferredMode)
+        }
```

Comment:
Nitpicky, but IMO this could be cleaner code and reduce some duplication.

```typescript
const toggleTheme = (type) => {
    let newTheme = type
    switch (type) {
        case .systemMode: {
             newTheme = preferredMode
             break;
        }
    }

    localStorage.setItem('theme', type)
    document...setAttribute('data-theme', theme);
    setTheme(theme)
}
```

Reply:
Makes sense - I've made this change